File: backend/src/main/java/com/project/entities/ProfessionalAvailability.java
package com.project.entities;

import java.time.LocalTime;

import com.project.enums.DayOfWeekEnum;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "professional_availability")
public class ProfessionalAvailability {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long availabilityId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "professional_id", nullable = false)
    private Professional professional;

    @Enumerated(EnumType.STRING)
    private DayOfWeekEnum dayOfWeek;

    @NotNull
    private LocalTime startTime;

    @NotNull
    private LocalTime endTime;
}

File: backend_dotnet/SafeMind.Core/Entities/ProfessionalAvailability.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using SafeMind.Core.Enums;

namespace SafeMind.Core.Entities
{
    [Table("professional_availability")]
    public class ProfessionalAvailability
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long AvailabilityId { get; set; }

        [Required]
        [ForeignKey(nameof(Professional))]
        [Column("professional_id")]
        public long ProfessionalId { get; set; }

        public Professional Professional { get; set; } = null!;

        [Required]
        public DayOfWeekEnum DayOfWeek { get; set; }

        [Required]
        public TimeSpan StartTime { get; set; }

        [Required]
        public TimeSpan EndTime { get; set; }
    }
}
